network/p2p/dnsaddr/resolve.go

```diff
@@ -35,8 +35,14 @@ func Iterate(initial multiaddr.Multiaddr, controller ResolveController, f func(d
 	if resolver == nil {
 		return errors.New("passed controller has no resolvers Iterate")
 	}
+	const maxHops = 100
```

Comment:
What made you choose this approach over a visited set?

Reply:
this is simpler and has zero allocation overhead